src/magic_folder/testing/web.py

```diff
@@ -227,6 +235,34 @@ def render_PUT(self, request):
             request.setResponseCode(http.OK)  # replaced/modified files
         return cap
 
+    def _add_entry_to_dir(self, request, dircap, segments):
+        """
+        Adds an entry to a mutable directory. Only handles a single-level
+        deep.
+        """
+        if len(segments) != 1:
+            raise Exception(
+                "Need exactly one path segment (got {})".format(len(segments))
+            )
+        dircap = request.postpath[0].decode("ascii")
+        if not dircap.startswith("URI:DIR2"):
+            raise Exception(
+                "Can't add entry to non-mutable directory '{}'".format(dircap)
+            )
+        try:
+            dir_raw_data = self.data[dircap]
+        except KeyError:
+            raise Exception(
+                "No directory for '{}'".format(dircap)
+            )
+
+        content_cap = request.content.read().decode("utf8")
```

Comment:
How is `content_cap` derived here?  I mean, `request.content` is basically the payload of request, right?  Is this not meant to be a "real-like" capability?

Reply:
I *believe* this is the same as the Tahoe-LAFS API (see, e.g., `scripts/tahoe_mv.py` around line 56). That said, this is a disadvantage of this whole testing strategy based around a "verified fake" that is trying to replicate the "real" Tahoe WebUI; the "verified" part is lacking, and since we're writing the fake as we go, it's definitely not complete :/ In an ideal world, the "verified fake" would come with Tahoe.

...however also we "can't" use any real-reactor stuff in our tests (or testtools + hypothesis don't work) so using the real underlying Tahoe code as the test-double can't be done either.